src/components/icon/icon.ts

```diff
@@ -0,0 +1,209 @@
+   * Examples:
+   *   'social:cake' -> ['social', 'cake']
+   *   'penguin' -> ['', 'penguin']
+   *   null -> ['', '']
+   */
+  private _splitIconName(iconName: string): [string, string] {
+    if (!iconName) {
+      return ['', ''];
+    }
+    const separatorIndex = this.svgIcon.indexOf(':');
+    if (separatorIndex == -1) {
+      return ['', iconName];
+    }
+    return [iconName.substring(0, separatorIndex), iconName.substring(separatorIndex + 1)];
+  }
+
+  ngOnChanges(changes: { [propertyName: string]: SimpleChange }) {
+    const changedInputs = Object.keys(changes);
+    // Only update the inline SVG icon if the inputs changed, to avoid unnecessary DOM operations.
+    if (changedInputs.indexOf('svgIcon') != -1 || changedInputs.indexOf('svgSrc') != -1) {
+      if (this.svgIcon) {
+        const [namespace, iconName] = this._splitIconName(this.svgIcon);
+        this._mdIconRegistry.getNamedSvgIcon(iconName, namespace).subscribe(
+            svg => this._setSvgElement(svg),
+            (err: any) => console.log(`Error retrieving icon: ${err}`));
+      } else if (this.svgSrc) {
+        this._mdIconRegistry.getSvgIconFromUrl(this.svgSrc).subscribe(
+            svg => this._setSvgElement(svg),
+            (err: any) => console.log(`Error retrieving icon: ${err}`));
+      }
+    }
+    if (this._usingFontIcon()) {
+      this._updateFontIconClasses();
+    }
+    this._updateAriaLabel();
+  }
+
+  ngOnInit() {
+    // Update font classes because ngOnChanges won't be called if none of the inputs are present,
+    // e.g. <md-icon>arrow</md-icon>. In this case we need to add a CSS class for the default font.
+    if (this._usingFontIcon()) {
+      this._updateFontIconClasses();
+    }
+  }
+
+  ngAfterContentChecked() {
+    // Update aria label here because it may depend on the projected text content.
+    // (e.g. <md-icon>home</md-icon> should use 'home').
+    this._updateAriaLabel();
+  }
+
+  private _updateAriaLabel() {
+      const ariaLabel = this._getAriaLabel();
+      if (ariaLabel) {
+        this._renderer.setElementAttribute(this._element.nativeElement, 'aria-label', ariaLabel);
+        this._changeDetectorRef.detectChanges();
+      }
+  }
+
+  private _getAriaLabel() {
+    // If the parent provided an aria-label attribute value, use it as-is. Otherwise look for a
+    // reasonable value from the alt attribute, font icon name, SVG icon name, or (for ligatures)
+    // the text content of the directive.
+    const label =
+        this.ariaLabelFromParent ||
+        this.alt ||
+        this.fontIcon ||
+        this._splitIconName(this.svgIcon)[1];
+    if (label) {
+      return label;
+    }
+    // The "content" of an SVG icon is not a useful label.
+    if (this._usingFontIcon()) {
+      const text = this._element.nativeElement.textContent;
+      if (text) {
+        return text;
+      }
+    }
+    // Warn here?
+    return null;
```

Comment:
We're holding off on logging aria warnings now because we want to be able to only output them in dev mode.